Add Laravel Forge extension

KevinBatdorf opened the pull request:
## Description

This adds a Laravel Forge "command center" type extension to let users control their Forge servers and sites.

https://forge.laravel.com/

**Features from Forge API**
- View site details
- Open in browser
- View .env/nginx file
- View deployment status
- Allows two accounts
- Trigger deploy script
- Reboot services

**Non-Forge API Features**
- Check site connectivity
- Open terminal session
- Copy IP address

## Type of change

- New extension

## Screencast

https://twitter.com/kevinbatdorf/status/1472654911930347520

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

pernielsentikaer commented:
Hi @KevinBatdorf 🔥 

Is this ready for review?

KevinBatdorf commented:
Hi @pernielsentikaer Yes, thanks!

pernielsentikaer reviewed (changes requested):
Hi @KevinBatdorf 🔥 

I gave it a quick spin and found one thing that we need to address before continuing.

If wrong credentials are entered, a nice toast is shown (which is correct) – what I miss here is that the loader stops, and it shows an empty array (Which trigger “No Results”) – can you look into this?  

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

KevinBatdorf commented:
Hey @pernielsentikaer I updated the extension and also added some changes I saw you mention on #634 which is based off this extension. 

I can't give you an API key unfortunately as I'm not affiliated with Laravel Forge, but if you think it's important I could reach out to them, or maybe even just share my personal token for a short while (lets chat on Slack if so).

pernielsentikaer commented on extensions/laravel-forge/src/Server.tsx:
```diff
@@ -0,0 +1,248 @@
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <PushAction title="Open Server Information" target={<SingleServerView server={server} sites={sites} />} />
```
Your’e missing an Icon here

pernielsentikaer commented on extensions/laravel-forge/src/Site.tsx:
```diff
@@ -0,0 +1,208 @@
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <PushAction title="Open Site Info" target={<SitesSingleView site={site} server={server} />} />
```
Your’e missing an Icon here

pernielsentikaer commented on extensions/laravel-forge/src/Site.tsx:
```diff
@@ -0,0 +1,208 @@
+            accessoryTitle="press to view"
+            actions={
+              <ActionPanel>
+                <PushAction title="Open .env File" target={<EnvironmentFile site={site} server={server} />} />
```
Your’e missing an Icon here

pernielsentikaer reviewed (changes requested):
Hi @KevinBatdorf 👋 🤛 

Awesome extension you’re doing here, I don’t personally use Laravel Forge, but it looks awesome with a lot of possibilities.

I only found a few places where icons are missing, besides that it looks pretty good to me. 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

KevinBatdorf commented:
Hey @pernielsentikaer I've added those Icons:

![Screen Shot 2022-01-16 at 2 58 39 PM](https://user-images.githubusercontent.com/1478421/149675875-88e37f5c-b01c-46ae-bc61-ea07fb52697e.png)

pernielsentikaer reviewed (approved):
Hi @KevinBatdorf 👋 

It looks awesome, and the new icon is looking SO good – I think we're ready to get this live!

I'm going to accept it and then will my colleague do a final review before it's finally merged.
